### PR TITLE
Aux Base Improvements

### LINKED ISSUE #24123: Miners can not use the shuttle console in the auxillary base
EDIT: FUCK

Okay so it says access denied when you try to send shuttle using the console at the base after you setup the shuttle landing zone

### PR DESCRIPTION
**Pull request by GunHog**
:cl: Gun Hog
add: The Aux Base console now controls turrets made by the construction console.
add: The Aux Base may now be dropped at a random location if miners fail to use the landing remote.
add: The mining shuttle may now dock at the Aux Base's spot once the base is dropped.
tweak: Removed access levels on the mining shuttle so it can be used at the public dock.
tweak: The Aux Base's turrets now fire through glass. Reminder that the turrets need to be installed outside the base for full damage.
add: Added a base construction console to Delta Station.
/:cl:

- The Base Management Console (BMC) may now set up its own landing zone
at random if mining has not prepared a drop zone.
- The BMC may now control the Perimeter Defense Turrets built by the
construction console. Their status and health may be checked from the
BMC as well.
- The PDTs now have slightly increased damage and fire through glass.
- Moved the BMC to its own file.
- Changed the path of the BMC in order to function properly as its own
computer type.
- The Aux Base's dock now doubles as a public mining shuttle dock once
the base is dropped and shuttle landing beacon set up. Controlled from
the BMC.
- Removed access requirements from the Box station mining shuttle to be
compabile with the public dock.
- Added a base construction console to Deltastation, moved the shuttle
beacon inside of the base, and moved a potted plant.

fixes https://github.com/tgstation/tgstation/issues/24123

screenshot 2017-02-15 14 27 56

### CODE REVIEW

**Comment by optimumtact** [comment]
Straight up buffs to the turrets that make it even more effective at killing lavaland fauna with zero justification.

**Comment by XDTM** [comment]
Justification being that for some reason any lavaland mob brushing against a wall will disintegrate it, so turrets have to kill a whole goliath before it gets close

**Comment by PKPenguin321** [comment]
That changelog is MASSIVE. Find a way to be more concise.

**Comment by Kromgar** [comment]
I waas on delta and i didn't see a aux base console and the shuttle beacon was outside of the base. 



@XDTM 

Fuckin for real everytime i see it dropped the base gets FUUUUCKED by basilisks and goliaths

**Comment by GunHog** [comment]
> Straight up buffs to the turrets that make it even more effective at killing lavaland fauna with zero justification.

I have been repeatedly testing base drops during my coding of this project. Many times, the base landed at or near a tunnel containing monsters which often destroyed the turrets and smashed the walls within seconds of the landing. The turrets have poor aiming logic and gleefully shot the glass portions of the base, causing breaches. I wanted players with glass components in their base to be able to use them without worrying about the turrets breaching them.

I can firmly say, backed by at least 10 test drops, that the turrets _sucked_, breached my own base, and shot me if I happened to be in the way. I do not want to discourage people using glass to liven up their base, and with my testing of the new beam turrets, I find that they are just decent enough to not be terrible. 

Although they probably will still wind up being destroyed in seconds. They work well if you back them up with your KA, though (and do not wander into the projectiles).

It seems I forgot about Pubby. I will fix that soon.

**Comment by GunHog** [comment]
**I cannot fix Pubby.** I need some help with this, as that map seems to be outdated and refuses to open:

image

**Comment by Repukan** [comment]
Why not clear the area before marking it for sending?

If you're desperate enough to send it randomly, why should this "last option" be safer?

**Comment by GunHog** [comment]
Clarify your statement, please. I do not know what you mean.

**Comment by Repukan** [comment]
1. If monsters in a marked area are going to be a problem, what's stopping science from asking the miners to clear out hostiles before docking in?

2. If you're sending it at random either from being desperate or bored, why are you getting turrets that minimize the danger of your recklessness? This just screams hugbox to me. Anything involving recklessly diving into lavaland should, as well, carry dangers that the person prepares for. Giving them turrets kill this. 

Ie. Sending a pod away in red alert to escape whatever on station carries a great risk because you could land on lava or near a tendril, so people tend to bring weapons or just submit to dying.

**Comment by Arianya** [comment]
I feel this is a positive change overall. The Aux base sees rare usage right now, so some buffs are warranted to get a feel for where the issue lies that creates that unpopularity

@GunHog How does this interact with megafauna? Probably don't want the base to be hotdropped near megafauna to give supporting fire.

**Comment by Repukan** [comment]
It started as a small zone for construction projects, then expanded to be sent to lavaland when it was finished, then into an aux base, now a console, turreted safe space?

If you don't put turrets in the mining station (which sees itself attacked by Goliath's and even megafauna when they spawn close enough) then what makes this off project so much more valuable that NT would spend into turrets?

It not seeing use might be, more likely than not, due to how short rounds have become. Making it a safe space will only serve as a contradiction to what the station is supposed to be (a metal death trap in spaaaaaaace) and indirectly buff antagonists that benefit from base building.

**Comment by SgtxRootbeer** [comment]
@repukan Because this isn't an emergency pod. This is a prepared, deliberate expedition to lavaland using purpose-built equipment and pods. 
An escape pod carries risk because it is a spur-of-the moment choice, built more as an afterthought for the station. 
Hell, I like this because, if there is a red-alert on the station, a person now has a choice, risk spending time making the base, or pod yourself.

**Comment by GunHog** [comment]
The turrets will automatically engage the Megafauna, which will enrage it. Said megafauna will promptly level the base and anyone unfortunate enough to be trapped inside. 

@Repukan The turrets provide a level of defense, but by no means is it safety. In same cases, they are a liability - especially with goliaths and legions. Goliaths are tanky and will often manage to make it to the turret and smash it. Legion, with their rapid skulls swarm, can overwhelm the turret unless it manages to hit the host. Alone, they can handle one, at most two mobs if intelligently placed. 

If left active, they can become a liability by attracting fauna that have not been aggroed - this happened a few times in my test as well. Goliaths one-hit walls by the way. 

What I want to see is people preparing equipment and possibly having staff that primarily stay inside the base and work. I want to see people bring fauna to the base and end up getting everyone inside killed. It LOOKS like a "safe space" on paper. People will build it to be one, so they feel 'safe' enough to want to **try.** 

I want people to use glass in their bases so they can look outside, no one smart would use turrets and glass together if I left them as is. Those turrets are not as strong as they seem, but strong enough to drive off a small number of mobs, if used well. The power controls at the command console are there for more cautions players to turn them off to avoid angering mobs, but to turn them back on when they are already attacking.

**Comment by Robustin** [comment]
I havent used this yet but how exactly do you "move turrets outside the base so that they do full damage"?

**Comment by SgtxRootbeer** [comment]
@robustin
I would assume you put the turret on a corner of the base, and wall around that corner instead of in that corner...
oox
oxx
xx*
Like this diagram for the bottom right side of the base.
O = empty space inside the base
X = wall
* = Turret.

**Comment by GunHog** [comment]
Like Sgt says, they should be exposed to the environment. They do not have to be in the CORNERS, but they should be outside, exposed to the low pressure to get that bonus. 

screenshot 2017-02-10 16 18 47

File: code/modules/mining/aux_base.dm [hunk 1]
```diff
@@ -0,0 +1,338 @@
+///Mining Base////
+
+/area/shuttle/auxillary_base
+	name = "Auxillary Base"
+	luminosity = 0 //Lighting gets lost when it lands anyway
+	#define BAD_ZLEVEL	1
```
Comment (AnturK) [inline code/modules/mining/aux_base.dm]: Move these to the top of the file please, the indentation doesn't really mean anything and let's stick to conventions.

File: code/modules/mining/aux_base.dm [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+/obj/machinery/computer/auxillary_base/New(location, obj/item/weapon/circuitboard/computer/shuttle/C)
+	..()
+	locator = new /obj/item/device/gps/internal/base(src)
+
+
+/obj/machinery/computer/auxillary_base/attack_hand(mob/user)
+	if(..(user))
+		return
+	src.add_fingerprint(usr)
```
Comment (AnturK) [inline code/modules/mining/aux_base.dm]: Remove redundant src. please.

File: code/modules/mining/aux_base.dm [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+	if(LAZYLEN(turrets))
+		if(href_list["turrets_power"])
+			for(var/PDT in turrets)
+				var/obj/machinery/porta_turret/aux_base/T = PDT
+				if(href_list["turrets_power"] == "on")
+					T.on = TRUE
+				else
+					T.on = FALSE
+		if(href_list["single_turret_power"])
+			var/obj/machinery/porta_turret/aux_base/T = locate(href_list["single_turret_power"])
```
Comment (AnturK) [inline code/modules/mining/aux_base.dm]: locate in turrets at least. Sanity checks for /ref spill would be nice too

File: code/modules/mining/aux_base.dm [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+	aux_base_console.set_mining_mode() //Lets the colony park the shuttle there, now that it has a dock.
+	user << "<span class='notice'>Mining shuttle calibration successful! Shuttle interface available at base console.</span>"
+	anchored = 1 //Locks in place to mark the landing zone.
+	playsound(src.loc, 'sound/machines/ping.ogg', 50, 0)
+
+/obj/structure/mining_shuttle_beacon/proc/clear_cooldown()
+	anti_spam_cd = 0
+
+/obj/structure/mining_shuttle_beacon/attack_robot(mob/user)
+	return (attack_hand(user)) //So borgies can help
```
Comment (AnturK) [inline code/modules/mining/aux_base.dm]: drop the ()

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/mining/aux_base.dm (hunk 3 above)
Comment (GunHog) [inline code/modules/mining/aux_base.dm]: I do not understand what you mean here

**Comment by GunHog** [comment]
Made requested changes except something about "/ref spill". I do not understand hrefs, so I do not know what this means.

**Comment by GunHog** [comment]
Change made

**Review by AnturK — APPROVED** [review]

**Pull request merged**